src/components/navigation/navigation.tsx

```diff
@@ -0,0 +1,221 @@
+  //
+  //  Private Properties/ State
+  //
+  //--------------------------------------------------------------------------
+
+  @State() logoSlotHasElements: boolean;
+
+  @State() navActionSlotHasElements: boolean;
+
+  @State() primaryContentCenterSlotHasElements: boolean;
+
+  @State() primaryContentEndSlotHasElements: boolean;
+
+  @State() primaryContentStartSlotHasElements: boolean;
+
+  @State() progressSlotHasElement: boolean;
+
+  @State() secondarySlotHasElements: boolean;
+
+  @State() tertiarySlotHasElements: boolean;
+
+  @State() userSlotHasElements: boolean;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  /** When navAction is true, emits when the displayed action selection changes.*/
+  @Event({ cancelable: false }) calciteNavActionSelect: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private clickHandler = () => {
+    this.calciteNavActionSelect.emit();
+  };
+
+  private handleUserSlotChange = (event: Event): void => {
+    if (this.isPrimaryLevel()) {
+      this.userSlotHasElements = slotChangeHasAssignedElement(event);
+    }
+  };
+
+  private handleLogoSlotChange = (event: Event): void => {
+    if (this.isPrimaryLevel()) {
+      this.logoSlotHasElements = slotChangeHasAssignedElement(event);
+    }
+  };
+
+  private handleContentStartSlotChange = (event: Event): void => {
+    if (this.isPrimaryLevel()) {
+      this.primaryContentStartSlotHasElements = slotChangeHasAssignedElement(event);
+    }
+  };
+
+  private handleContentEndSlotChange = (event: Event): void => {
+    if (this.isPrimaryLevel()) {
+      this.primaryContentEndSlotHasElements = slotChangeHasAssignedElement(event);
+    }
+  };
+
+  private handleContentCenterSlotChange = (event: Event): void => {
+    if (this.isPrimaryLevel()) {
+      this.primaryContentCenterSlotHasElements = slotChangeHasAssignedElement(event);
+    }
+  };
+
+  private handleSecondarySlotChange = (event: Event): void => {
+    this.secondarySlotHasElements = slotChangeHasAssignedElement(event);
+  };
+
+  private handleTertiarySlotChange = (event: Event): void => {
+    this.tertiarySlotHasElements = slotChangeHasAssignedElement(event);
+  };
+
+  private handleMenuActionSlotChange = (event: Event): void => {
+    if (this.isPrimaryLevel()) {
+      this.navActionSlotHasElements = slotChangeHasAssignedElement(event);
+      if (this.navActionSlotHasElements) {
+        this.navAction = false;
+      }
+    }
+  };
+
+  private handleProgressSlotChange = (event: Event): void => {
+    if (this.isPrimaryLevel()) {
+      this.progressSlotHasElement = slotChangeHasAssignedElement(event);
+    }
+  };
+
+  private isPrimaryLevel = (): boolean => {
```

Comment:
This can be a class function instead of an assigned bound function. `this` is already set where it's called.

Reply:
the reason for having this is to avoid having different scope for `this` when used inside `onSlotchange` event handlers.

Reply:
All of the `onSlotchange` event handlers already have `this` bound to the instance. Hence why we don't need it here.